Microsoft Sharepoint 3.2 (#1365)
mulesoft/docs-connectors: sharepoint/3.2/modules/ROOT/pages/sharepoint-connector-reference.adoc

= SharePoint Connector 3.2 Reference
:page-aliases: connectors::sharepoint/sharepoint-connector-reference.adoc

Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]

Anypoint Connector for SharePoint enables you to connect to a Microsoft SharePoint application.

> Release Notes: xref:release-notes::connector/sharepoint-connector-release-notes-mule-4.adoc[Microsoft SharePoint Connector Release Notes]

Reviewer: Delete

| Port a| Number |  Port number for the proxy server. |  |
| Username a| String |  Username to use for authenticating against the proxy server. |  |
| Password a| String |  Password to use for authenticating against the proxy server. |  |
| NTLM Domain a| String |  Domain to use for authenticating against the proxy server. |  |
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
| Consumer Key a| String |  OAuth consumerKey registered with the service provider. |  | x
| Consumer Secret a| String |  OAuth consumerSecret registered with the service provider. |  | x
| Authorization Url a| String |  Authorization endpoint URL of the service provider. |  https://login.microsoftonline.com/{tenant}/oauth2/v2.0/authorize |
| Access Token Url a| String |  Access token endpoint URL of the service provider. |  https://login.microsoftonline.com/{tenant}/oauth2/v2.0/token |
| Scopes a| String |  OAuth scopes requested during the dance. If not provided, it will default to those in the annotation. |  |

> | Resource Owner Id a| String |  The resourceOwnerId that each component uses if not referenced otherwise. |  |

Reviewer: ```suggestion
| Resource Owner Id a| String |  The resource owner ID that each component uses if it is not otherwise referenced. |  |
```

> | Before a| String |  Name of a flow executed right before starting the OAuth dance. |  |

Reviewer: ```suggestion
| Before a| String |  Name of the flow that is executed immediately before starting the OAuth dance. |  |
```

> | After a| String |  Name of a flow executed right after an accessToken is received. |  |

Reviewer: ```suggestion
| After a| String |  Name of the flow that is executed immediately after an access token is received. |  |
```

> | Listener Config a| String |  A reference to a <http:listener-config /> used to create the listener that will catch the access token callback endpoint. |  | x

Reviewer: ```suggestion
| Listener Config a| String | Reference to a <http:listener-config /> used to create the listener that receives the access token callback endpoint. |  | x
```

| Callback Path a| String |  Path of the access token callback endpoint.|  | x

> | Authorize Path a| String |  Path of the local http endpoint that triggers the OAuth dance.|  | x

Reviewer: ```suggestion
| Authorize Path a| String |  Path of the local HTTP endpoint that triggers the OAuth dance.|  | x
```

> | External Callback Url a| String |  If the callback endpoint is behind a proxy or should be accessed through a non direct URL, use this parameter to tell the OAuth provider the URL it should use to access the callback. |  |

Reviewer: ```suggestion
| External Callback Url a| String |  If the callback endpoint is behind a proxy or accessed through a non direct URL, use this parameter to specify the OAuth provider that the URL should use to access the callback. |  |
```

> | Object Store a| String |  A reference to the object store that should be used to store each resource owner ID's data. If not specified, runtime will automatically provision the default one. |  |

Reviewer: ```suggestion
| Object Store a| String |  A reference to the object store to use to store each resource owner ID's data. If a value is not specified, Mule automatically provisions the default object store. |  |
```

* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileDeny]]
=== File Deny
`<sharepoint:file-deny>`

Denies approval for a file that was submitted for content approval. Only documents in lists that are enabled for content approval can be denied.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use | | x
| File Server Relative Url a| String | The path of the file to be denied |  | x
| Comment a| String | Additional comment for this deny operation |  |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileGetContent]]
=== File Get Content
`<sharepoint:file-get-content>`

Retrieves content of a file in SharePoint.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use. | | x

> | File Server Relative Url a| String | Target file to retrieve its content from SharePoint. |  | x

Reviewer: ```suggestion
| File Server Relative Url a| String | Target file that contains the content to retrieve from SharePoint. |  | x
```

| Target Variable a| String |  The name of a variable that stores the operation's output. |  |
| Target Value a| String |  An expression that evaluates against the operation's output. The outcome of this expression is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |Binary
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileMove]]
=== File Move
`<sharepoint:file-move>`

> Executes an HTTP POST against the Sharepoint API to move a file. The parameter fileServerRelativeUrl needs to point to an existing file in SharePoint. The parameter newFileServerRelativeUrl should be a path to where the file should be moved. If this is a directory, an error will be thrown since it should be a new file path. It can have the same name as the old file or it can have a new name. In order to use this operation as just a rename, specify the same path in both parameters and use a different file name in newFileServerRelativeUrl.

Reviewer: ```suggestion
Executes an HTTP POST against the Sharepoint API to move a file. The `fileServerRelativeUrl` parameter must point to an existing file in SharePoint. The `newFileServerRelativeUrl` parameter must be the path for where to move the file. If the provided path is to a directory, an error will be thrown since it should be a new file path. The new file can have the same name as the old file or it can have a new name. 

To use this operation as a rename operation, specify the same path in both parameters and use a different file name in the `newFileServerRelativeUrl` parameter.
```

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|======================
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use | | x
| File Server Relative Url a| String |  Path to the source file |  | x
| New File Server Relative Url a| String |  Path to the destination file |  | x

> | Flag a| Number |

Reviewer: ```suggestion
| Flag a| Number a|
```

* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileRecycle]]
=== File Recycle
`<sharepoint:file-recycle>`

Moves the file to the Recycle Bin and returns the identifier of the new Recycle Bin item.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use. | | x
| File Server Relative Url a| String | The path of the file to be recycled. |  | x
| Target Variable a| String |  The name of a variable that stores the operation's output |  |
| Target Value a| String |  An expression that evaluates against the operation's output. The outcome of this expression is stored in the target variable. |  #[payload] |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |String
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileUndoCheckOut]]
=== File Undo Check Out
`<sharepoint:file-undo-check-out>`

> If a file has been checkout, this operation can revert the checkout state.

Reviewer: ```suggestion
If a file is checked out, this operation can revert the checkout state.
```

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use | | x

> | File Server Relative Url a| String | The path of the file to be reverted |  | x

Reviewer: ```suggestion
| File Server Relative Url a| String | The path of the file to revert |  | x
```

| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[fileUnpublish]]
=== File Unpublish
`<sharepoint:file-unpublish>`

> Removes the file from content approval or unpublish a major version.

Reviewer: ```suggestion
Removes the file from content approval or unpublishes a major version.
```

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use | | x

> | File Server Relative Url a| String | The path of the file to be unpublished |  | x

Reviewer: ```suggestion
| File Server Relative Url a| String | Path of the file to unpublish |  | x
```

| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |<<SharepointFolder>>
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[folderDelete]]
=== Folder Delete
`<sharepoint:folder-delete>`

Deletes a folder from a Document list

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use | | x
| Url a| String |  Server relative URL of the folder to delete. For example, `/Shared Documents/new folder`, deletes `new folder` from the `Shared Documents` list. |  | x
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== Associated Configurations
* <<sharepoint-online>>
* <<sharepoint-onpremise>>

==== Throws
* SHAREPOINT:RETRY_EXHAUSTED
* SHAREPOINT:UNKNOWN
* SHAREPOINT:CONNECTIVITY

[[folderQuery]]
=== Folder Query
`<sharepoint:folder-query>`

> Retrieves all folders that matches the specified criteria

Reviewer: ```suggestion
Retrieves all folders that match the specified criteria
```
